src/server/snap.rs

```diff
@@ -86,41 +85,44 @@ fn send_snap(mgr: SnapManager, addr: SocketAddr, data: ConnData) -> Result<()> {
     let snap = data.msg.get_raft().get_message().get_snapshot();
     let key = try!(SnapKey::from_snap(&snap));
     mgr.wl().register(key.clone(), SnapEntry::Sending);
-    let snap_file = box_try!(mgr.rl().get_snap_file(&key, true));
-    defer!({
```

Comment:
It gets a compile error to use a immutable refer of snap_file in defer, and then using `&mut snap_file` later in `io::copy`.